pkg/controller/dataimportcron-controller.go

```diff
@@ -127,8 +127,11 @@ func (r *DataImportCronReconciler) Reconcile(ctx context.Context, req reconcile.
 	if !shouldReconcile || err != nil {
 		return reconcile.Result{}, err
 	}
-	if err := r.initCron(ctx, dataImportCron); err != nil {
-		return reconcile.Result{}, err
+	// Cron job and initial job are not initialized if no schedule, allowing external source update trigger
+	if dataImportCron.Spec.Schedule != "" {
+		if err := r.initCron(ctx, dataImportCron); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
The check should be inside `initCron()` as it may do other things except initializing CronJob and initial Job.

Reply:
fixed